recipes/sharedtags-greedyview.lua

```diff
@@ -0,0 +1,195 @@
+        IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+        FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+        AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+        LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+        FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+        IN THE SOFTWARE.
+    ]]
+}
+
+-- Add a signal for each new screen, which just listens for the remove
+-- event, and moves over all tags when it happens.
+awful.screen.connect_for_each_screen(function(s)
+    -- When the screen is removed, all tags need to be moved over to an existing
+    -- screen. If they are not, accessing the tags will result in an error. It
+    -- doesn't make sense to fix the error, since clients on the now-hidden tags
+    -- will automatically be moved to a tag on a visible screen.
+    s:connect_signal("removed",function()
+        -- The screen to move the orphaned tags to.
+        local newscreen = capi.screen.primary
+        -- The currently selected tags on that screen.
+        local seltags = newscreen.selected_tags
+
+        -- Move over all tags to an existing screen.
+        for _,tag in ipairs(s.tags) do
+            sharedtags.movetag(tag, newscreen)
+        end
+
+        -- Restore the viewed tags on the new screen.
+        for i,tag in ipairs(seltags) do
+            if i == 1 then
+                tag:view_only()
+            else
+                awful.tag.viewtoggle(tag)
+            end
+        end
+    end)
+end)
+
+--- Create new tag objects.
+-- The first tag defined for each screen will be automatically selected.
+-- @tparam table def A list of tables with the optional keys `name`, `layout`
+-- and `screen`. The `name` value is used to name the tag and defaults to the
+-- list index. The `layout` value sets the starting layout for the tag and
+-- defaults to the first layout. The `screen` value sets the starting screen
+-- for the tag and defaults to the first screen. The tags will be sorted in this
+-- order in the default taglist.
+-- @treturn table A list of all created tags. Tags are assigned numeric values
+-- corresponding to the input list, and all tags with non-numerical names are
+-- also assigned to a key with the same name.
+-- @usage local tags = sharedtags(
+--   -- "main" is the first tag starting on screen 2 with the tile layout.
+--   { name = "main", layout = awful.layout.suit.tile, screen = 2 },
+--   -- "www" is the second tag on screen 1 with the floating layout.
+--   { name = "www" },
+--   -- Third tag is named "3" on screen 1 with the floating layout.
+--   {})
+-- -- tags[2] and tags["www"] both refer to the same tag.
+function sharedtags.new(def)
+    local tags = {}
+
+    for i,t in ipairs(def) do
+        tags[i] = awful.tag.add(t.name or i, {
+            screen = (t.screen and t.screen <= capi.screen.count()) and t.screen or capi.screen.primary,
```

Comment:
What's up with `t.screen <= capi.screen.count()`? That seems (a) pointless (if screens are numbers) or (b) broken (if screen objects are used, which I think is actually the case).